spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/BlockingQueueConsumer.java

```diff
@@ -483,8 +518,11 @@ private void attemptPassiveDeclarations() {
 				if (logger.isWarnEnabled()) {
 					logger.warn("Failed to declare queue:" + queueName);
 				}
+				if (!this.connection.isOpen()) {
```

Comment:
Polish looks good.

Rationale; my main reason for checking the connection is because the underlying channel closes when a declaration fails (e.g. due to a 404 NOT_FOUND), and it is refreshed by the `CCF.CCIH` by catching the `InvocationTargetException` so we'd actually be testing a different channel.

Of course, if the error was due to a lost connection and the channel `isOpen()` then it means the connection/channel was successfully refreshed. If the channel could not be refreshed, it would be `null` and the proxy would return false anyway, so that would be ok.

I was originally looking at adding code to only retry if we got a 404 (or maybe a 405) but because the refresh logic is in the catch block we might not actually get the original exception - we might get a new exception from the `createBareChannel`.

Now I think about it some more (thanks to your question). I think it's better to test the channel because the connection we captured during `start()` could possibly be a different connection to the one the exception actually occurred on (due to the refresh logic), especially when declaring multiple queues.

Pushed

I will apply your polish and push another change